RFS/src/main/java/com/rfs/common/http/RestClient.java

```diff
@@ -0,0 +1,185 @@
+                                       IRfsContexts.IRequestContext context) {
+        var convertedHeaders = additionalHeaders.entrySet().stream().collect(Collectors
+            .toMap(Map.Entry::getKey, e -> List.of(e.getValue())));
+        return asyncRequest(method, path, body, convertedHeaders, context);
+    }
+
+
+    public Mono<Response> asyncRequest(HttpMethod method, String path, String body, Map<String, List<String>> additionalHeaders,
+                                       @Nullable IRfsContexts.IRequestContext context) {
+        Map<String, List<String>> headers = new HashMap<>(Map.of(
+            "User-Agent", List.of("RfsWorker-1.0"),
+            "Host", List.of(getHostHeader(connectionContext.url))
+        ));
+        if (body != null) {
+            headers.put("Content-Type", List.of("application/json"));
+        }
+        if (additionalHeaders != null) {
+            additionalHeaders.forEach((key, value) -> {
+                    if (headers.containsKey(key.toLowerCase())) {
+                        headers.put(key.toLowerCase(), value);
+                    } else {
+                        headers.put(key, value);
+                    }
+                });
+        }
+
+        var responseGetter = authTransformer.transform(method.name(), path, headers, Mono.justOrEmpty(body))
+            .flatMap(transformedRequest -> client.doOnRequest(addSizeMetricsHandlers(context))
+                .headers(h -> transformedRequest.headers.forEach(h::add))
+                .request(method)
+                .uri("/" + path)
+                .send(transformedRequest.body
+                    .map(mapBody -> Unpooled.wrappedBuffer(mapBody.getBytes(StandardCharsets.UTF_8)))
+                )
+                .responseSingle(
+                    (response, bytes) -> bytes.asString()
+                        .singleOptional()
+                        .map(bodyOp -> new Response(
+                            response.status().code(),
+                            bodyOp.orElse(null),
+                            response.status().reasonPhrase(),
+                            extractHeaders(response.responseHeaders())
+                        ))
+                ));
+        if (context != null) {
+            return responseGetter.doOnError(t -> context.addTraceException(t, true))
+                .doFinally(r -> context.close());
+        }
+        return responseGetter;
+    }
+
+    private Map<String, String> extractHeaders(HttpHeaders headers) {
+        return headers.entries().stream()
+            .collect(java.util.stream.Collectors.toMap(
+                Map.Entry::getKey,
+                Map.Entry::getValue,
+                (v1, v2) -> v1 + "," + v2
+            ));
+    }
+
+    public Response get(String path, IRfsContexts.IRequestContext context) {
+        return getAsync(path, context).block();
+    }
```

Comment:
Design nitpick: I'd rather we only expose have async methods with no alternatives, e.g. getAsync(...) -> get(...).  `.block()` is trivial to do inline and I'd prefer that we default to async behavior

Reply:
Github is making this look like it's all new code whereas i just moved the directory here. I'll update the commit history to reflect it's just a file move

Reply:
Oh thanks for the context, fine with me either way.